### PR TITLE
[plugin.video.massengeschmack] 2.2.0

### PR DESCRIPTION
**Pull request by phoerious**
### Description
Kodi client for the Massengeschmack.tv VoD portal.

The plugin already exists in the add-on repositories, but has not yet been added to this Git repository. 

Original URL is https://github.com/phoerious/massengeschmack-xbmc/releases/tag/v2.1.0. This is a subtree merge into a fork of your repository.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.massengeschmack/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.gitignore export-ignore
```
Comment (enen92) [inline plugin.video.massengeschmack/.gitattributes]: please drop this file from the pull-request

File: plugin.video.massengeschmack/.gitignore [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# OS-specific files
```
Comment (enen92) [inline plugin.video.massengeschmack/.gitignore]: this one too

Comment (phoerious) [inline plugin.video.massengeschmack/.gitignore]: That's exactly the reason why I used to do ZIP pulls. I need my make-release.sh script, but I'm pretty sure you don't want to have it in the repository. That's why the .gitattributes file is there which makes sure neither make-release.sh, nor .gitignore nor the .gitattributes file itself is in the release.

File: plugin.video.massengeschmack/resources/lib/datasource.py [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+        self.id           = -9999
+        self.moduleName   = 'live'
+        self.sortOrder    = 600
+        self.showMetaData = {
+            'Title'    : ADDON.getLocalizedString(30270),
+            'Country'  : ADDON.getLocalizedString(30202),
+            'Plot'     : ADDON.getLocalizedString(30272)
+        }
+        self.bannerPath = ADDON_BASE_PATH + '/resources/media/banner-massengeschmack-20160220.png'
+        self.fanartPath = ADDON_BASE_PATH + '/resources/media/fanart-massengeschmack-20160220.jpg'
```
Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/datasource.py]: os

File: plugin.video.massengeschmack/resources/lib/datasource.py [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+
+            metaData  = {
+                'Title'     : streamName,
+                'Date'      : date,
+                'Plot'      : plot
+            }
+
+            if mode == 'live' or mode == 'recording':
+                listUrl = resources.lib.assembleListURL(self.moduleName, playStream=i['showid'], streamName=streamName,
+                                                        iconImage=iconImage, metaData=json.dumps(metaData))
```
Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/datasource.py]: iconImage is deprecated, see the previous comment

File: plugin.video.massengeschmack/resources/lib/listing.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        Set the view mode of the current listing.
+        
+        @type id: int
+        @param id: the view mode ID from the current skin
+        """
+        xbmcplugin.setContent(ADDON_HANDLE, self.__source.getContentMode())
+        xbmc.executebuiltin('Container.SetViewMode(' + str(id) + ')')
+    
+    def __addDir(self, listItem):
+        xbmcListItem = xbmcgui.ListItem(listItem.getData('name'), iconImage=listItem.getData('thumbnail'),
```
Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/listing.py]: same

Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/listing.py]: Also please set the `mediatype` property for all listitems that provide one of these: `"video", "movie", "tvshow", "season", "episode" or "musicvideo"`. See setInfo documentation:

https://codedocs.xyz/AlwinEsch/kodi/group__python__xbmcgui__listitem.html#ga0b71166869bda87ad744942888fb5f14

File: plugin.video.massengeschmack/resources/lib/listing.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                                        thumbnailImage=listItem.getData('thumbnail'))
+        xbmcListItem.setInfo(type='video', infoLabels=listItem.getData('metaData'))
+        xbmcListItem.setProperty('fanart_image', listItem.getData('fanart'))
+        # fix for XBMC4Xbox
+        if not IS_XBOX:
```
Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/listing.py]: same

File: plugin.video.massengeschmack/addon.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        
+        <summary lang="en">Massengeschmack video portal</summary>
+        <summary lang="de">Massengeschmack-Video-Portal</summary>
+        
+        <description lang="en">Massengeschmack is a German online video network producing magazines like Fernsehkritik-TV, Pantoffel-TV and Pressesch(l)au.</description>
+        <description lang="de">Massengeschmack ist ein deutsches Online-Video-Netzwerk, das Magazine wie Fernsehkritik-TV, Pantoffel-TV und Pressesch(l)au produziert.</description>
+        
+        <disclaimer lang="en">This is not an official add-on and in no way affiliated with Massengeschmack</disclaimer>
+        <disclaimer lang="de">Dies ist kein offizielles Add-on und in keiner Weise an Massengeschmack angegliedert.</disclaimer>
+    </extension>
```
Comment (enen92) [inline plugin.video.massengeschmack/addon.xml]: please also create the assets tag and move both the icon.png and the fanart.jpg to the resources folder:

See (addon-structure for krypton): http://kodi.wiki/view/Add-on_structure
See (addon.xml krypton): http://kodi.wiki/view/addon.xml

File: plugin.video.massengeschmack/.gitignore (hunk 2 above)
Comment (phoerious) [inline plugin.video.massengeschmack/.gitignore]: Wow, you guys really make it as hard as possible for plugin authors to match your standards.

When I merge my stuff into the pull request repository, I have to add all files to the index and squash them into a commit. Having at least a .gitignore makes sure I don't commit anything that shouldn't be there. Is it really so bad to have at least that in the repository?

Don't you guys do a git export from this repository anyways? Then all that stuff would be gone automatically.

Comment (enen92) [inline plugin.video.massengeschmack/.gitignore]: But they will still be pulled and be present on the github repository for no reason. Same for specific IDE files. What makes it so hard for you to remove them from the branch you PR from?

**Comment by phoerious** [comment]
I updated the plugin according to your change requests and took the chance to make some more changes and bundle a 2.2.0 release.

File: plugin.video.massengeschmack/addon.xml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+<addon id="plugin.video.massengeschmack" name="Massengeschmack" version="2.2.0" provider-name="phoerious">
+    <requires>
+        <import addon="xbmc.python" version="2.25.0"/>
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <language>en_GB de_DE</language>
```
Comment (enen92) [inline plugin.video.massengeschmack/addon.xml]: this should be kept as "en de". The changes I requested are only applied for the language files (attributes for the summary, description and disclaimer).

**Comment by enen92** [comment]
@phoerious thanks for adjusting. Please check my last comment for the plugin to be merged.

File: plugin.video.massengeschmack/.gitignore (hunk 2 above)
Comment (phoerious) [inline plugin.video.massengeschmack/.gitignore]: I just feel like this whole process produces more work than necessary. I have to copy over all the stuff from the main repository and create a new commit. Having at least a .gitignore would make sure that no IDE files and other unwanted stuff ends up in that commit by accident.

File: plugin.video.massengeschmack/addon.xml (hunk 8 above)
Comment (phoerious) [inline plugin.video.massengeschmack/addon.xml]: Done.

**Comment by enen92** [comment]
Thanks

**Pull request merged**